src/main/java/com/powsybl/openloadflow/ac/ShuntVoltageControlOuterLoop.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.ac;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoop;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoopContext;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoopStatus;
+import com.powsybl.openloadflow.network.AbstractDiscreteVoltageControl;
+import com.powsybl.openloadflow.network.ShuntVoltageControl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+public class ShuntVoltageControlOuterLoop implements OuterLoop {
```

Comment:
Are you sure we need an outerloop? I do not see like for transformer a first step with continuous variable for susceptance then rounding and then we re-run with rounded values?

Reply:
Yes I do, but the global design of this PR was bad, so I have made some fixes to fit with what we do for transformers. It looks better, even if I think that we can improve the way we model shunts, maybe in two equations max by bus. I have made some tests on real test cases (Elia and Mavir) and for both I have an issue that says that I have too many variables for the number of equations directly in the first iteration:
For Elia:
`
09:05:37.442 [ForkJoinPool.commonPool-worker-3] ERROR  c.p.openloadflow.ac.nr.NewtonRaphson - com.powsybl.commons.PowsyblException: Expected to have same number of equations (1630) and variables (1631)
com.powsybl.commons.PowsyblException: Expected to have same number of equations (1630) and variables (1631)
`
Any idea?

Reply:
Fixed, it was a serious bug in the network loading. A bus was both controlled by a generator and shunts, but it was not detected during the loading. Now, it works on both real networks.